File: src/main/java/com/github/chMatvey/service/DroneService.java
package com.github.chMatvey.service;

import com.github.chMatvey.model.Drone;
import com.github.chMatvey.web.request.DroneRegisterRequest;
import com.github.chMatvey.web.response.DroneRegisterResponse;
import com.github.chMatvey.web.response.DroneResponse;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.transaction.Transactional;

import java.util.List;

import static com.github.chMatvey.model.DroneState.IDLE;

@ApplicationScoped
public class DroneService {
    public List<DroneResponse> availableDrones() {
        return Drone.available().stream()
                .map(DroneResponse::create)
                .toList();
    }

    @Transactional
    public DroneRegisterResponse registerDrone(DroneRegisterRequest request) {
        Drone drone = Drone.builder()
                .serialNumber(request.serialNumber())
                .model(request.model())
                .weightLimit(request.weightLimit())
                .batteryCapacity(100)
                .state(IDLE)
                .build();

        drone.persist();

        return new DroneRegisterResponse(drone.getId());
    }
}

File: src/main/kotlin/com/github/chMatvey/services/DroneService.kt
package com.github.chMatvey.services

import com.github.chMatvey.models.Drone
import com.github.chMatvey.services.dto.request.DroneRegisterRequest
import com.github.chMatvey.services.dto.response.DroneRegisterResponse

interface DroneService {
    suspend fun getAvailableDrones(): List<Drone>

    suspend fun getDroneById(droneId: Int): Drone?

    suspend fun registerDrone(request: DroneRegisterRequest): DroneRegisterResponse
}